# What is your Name and E-mail

NAME: DUBBAKA SRIKANTH                                                                                                 
EMAIL: srikanthdubbaka123@gmail.com

In [1]:
# create numpy in 5 builtin methods. 
#import numpy
import numpy as np
#created list
List=[0,1,2,3,4,5,6,7,8,9,10]
print("1.The list is:")
print(List)

#array of list
array= np.array(List)
print("2.array of list is:")
print(array)

#dtype=float
arrayis = np.array(List,dtype="float")
print("3.dtype =float--array is:")
print(arrayis)

#array-dtype
print("4.data type of array is:")
print(array.dtype)

#list is in dtype=int
arrayis = np.array(List,dtype="int")
print("5.The array of dtype=int is:")
print(arrayis)

#find array of alpha
a=['a','b','c']
b = np.array(a)
print("6.The array of alphabets(a) is:")
print(b)
print("7.data type of array of alphabets is:")
print(b.dtype)



1.The list is:
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
2.array of list is:
[ 0  1  2  3  4  5  6  7  8  9 10]
3.dtype =float--array is:
[ 0.  1.  2.  3.  4.  5.  6.  7.  8.  9. 10.]
4.data type of array is:
int32
5.The array of dtype=int is:
[ 0  1  2  3  4  5  6  7  8  9 10]
6.The array of alphabets(a) is:
['a' 'b' 'c']
7.data type of array of alphabets is:
<U1


In [2]:
# perform 4 operations on matrices
import numpy as np
# define a Python matrix with data elements in it
list = [[ 'sri', 1, 2, 3],
        ["raj", 4, 5, 6],
        [7, "raghu", 8, 9]]
print("1.Python matrix with data elements in list is:")
print(list)
# define the length of the list

print("2.length of list:")
print(len(list))

print("3.add two matrices using Numpy")
a = np.array([[1, 2], [3, 4]])
b = np.array([[5, 6], [7, 8]])
c = np.add(a, b)
print(c)
print("4.subtract two matrices using Numpy")
a = np.array([[1, 2], [3, 4]])
b = np.array([[5, 6], [7, 8]])
c = np.subtract(a, b)
print(c)
 

 

1.Python matrix with data elements in list is:
[['sri', 1, 2, 3], ['raj', 4, 5, 6], [7, 'raghu', 8, 9]]
2.length of list:
3
3.add two matrices using Numpy
[[ 6  8]
 [10 12]]
4.subtract two matrices using Numpy
[[-4 -4]
 [-4 -4]]


In [3]:
#Inverse of matrix
import numpy as np

a = np.array([[1, 2, 3], [1, 3, 3], [1, 2, 4]])
invofa= np.linalg.inv(a)

print(invofa)

[[ 6. -2. -3.]
 [-1.  1.  0.]
 [-1.  0.  1.]]


In [4]:
# write a custom operation function and verctorize it. 
# be careful while intializing applying on big matrices
import numpy as np
def func1(a, b):
    if a > b:
        return a - b
    else:
        return a + b
vfun = np.vectorize(func1)
x=vfun([1, 2, 3, 4], 5)
print(x)

[6 7 8 9]


In [5]:
# import numpy and pandas
import pandas as pd
import numpy as np

In [6]:
# import data into dataframes
company = pd.read_csv(r'G:\company.csv',delimiter='\t')
rounds = pd.read_csv(r'G:\rounds2.csv',encoding='iso-8859-1')
# we imported two files rounds and company. we have 2 dataframes

In [7]:
# convert rounds dataframe into numpy array
company.to_numpy()

array([['/Organization/-Fame', '#fame', 'http://livfame.com', ...,
        'Mumbai', 'Mumbai', nan],
       ['/Organization/-Qounter', ':Qounter', 'http://www.qounter.com',
        ..., 'DE - Other', 'Delaware City', '04-09-2014'],
       ['/Organization/-The-One-Of-Them-Inc-', '(THE) ONE of THEM,Inc.',
        'http://oneofthem.jp', ..., nan, nan, nan],
       ...,
       ['/Organization/Ã\x81Eron', 'Ã\x81ERON', 'http://www.aeron.hu/',
        ..., nan, nan, '01-01-2011'],
       ['/Organization/ÔAsys-2', 'Ôasys', 'http://www.oasys.io/', ...,
        'SF Bay Area', 'San Francisco', '01-01-2014'],
       ['/Organization/Ä°Novatiff-Reklam-Ve-Tanä±Tä±M-Hizmetleri-Tic',
        'İnovatiff Reklam ve Tanıtım Hizmetleri Tic',
        'http://inovatiff.com', ..., nan, nan, nan]], dtype=object)

In [8]:
# convert rounds dataframe into numpy array
rounds.to_numpy()

array([['/organization/-fame',
        '/funding-round/9a01d05418af9f794eebff7ace91f638', 'venture',
        'B', '05-01-2015', 10000000.0],
       ['/ORGANIZATION/-QOUNTER',
        '/funding-round/22dacff496eb7acb2b901dec1dfe5633', 'venture',
        'A', '14-10-2014', nan],
       ['/organization/-qounter',
        '/funding-round/b44fbb94153f6cdef13083530bb48030', 'seed', nan,
        '01-03-2014', 700000.0],
       ...,
       ['/organization/ã\x81eron',
        '/funding-round/59f4dce44723b794f21ded3daed6e4fe', 'venture',
        'A', '01-08-2014', nan],
       ['/ORGANIZATION/Ã\x94ASYS-2',
        '/funding-round/35f09d0794651719b02bbfd859ba9ff5', 'seed', nan,
        '01-01-2015', 18192.0],
       ['/organization/ä°novatiff-reklam-ve-tanä±tä±m-hizmetleri-tic',
        '/funding-round/af942869878d2cd788ef5189b435ebc4', 'grant', nan,
        '01-10-2013', 14851.0]], dtype=object)

## print basic imformation like shape,info(company)

In [9]:
company

permalink  \
0                                    /Organization/-Fame   
1                                 /Organization/-Qounter   
2                    /Organization/-The-One-Of-Them-Inc-   
3                                  /Organization/0-6-Com   
4                         /Organization/004-Technologies   
...                                                  ...   
66363  /Organization/Zznode-Science-And-Technology-Co...   
66364                          /Organization/Zzzzapp-Com   
66365                               /Organization/ÁEron   
66366                             /Organization/ÔAsys-2   
66367  /Organization/Ä°Novatiff-Reklam-Ve-Tanä±Tä±M-H...   

                                                name  \
0                                              #fame   
1                                           :Qounter   
2                             (THE) ONE of THEM,Inc.   
3                                            0-6.com   
4                                   004 Technologies   
...                                              ...   
66363                  ZZNode Science and Technology   
66364                          Zzzzapp Wireless ltd.   
66365                                         ÁERON   
66366                                         Ôasys   
66367  İnovatiff Reklam ve Tanıtım Hizmetleri Tic   

                            homepage_url  \
0                     http://livfame.com   
1                 http://www.qounter.com   
2                    http://oneofthem.jp   
3                     http://www.0-6.com   
4      http://004gmbh.de/en/004-interact   
...                                  ...   
66363              http://www.zznode.com   
66364             http://www.zzzzapp.com   
66365               http://www.aeron.hu/   
66366               http://www.oasys.io/   
66367               http://inovatiff.com   

                                           category_list     status  \
0                                                  Media  operating   
1      Application Platforms|Real Time|Social Network...  operating   
2                                      Apps|Games|Mobile  operating   
3                                            Curated Web  operating   
4                                               Software  operating   
...                                                  ...        ...   
66363                                Enterprise Software  operating   
66364        Advertising|Mobile|Web Development|Wireless  operating   
66365                                                NaN  operating   
66366  Consumer Electronics|Internet of Things|Teleco...  operating   
66367                 Consumer Goods|E-Commerce|Internet  operating   

      country_code state_code                 region           city  \
0              IND         16                 Mumbai         Mumbai   
1              USA         DE             DE - Other  Delaware City   
2              NaN        NaN                    NaN            NaN   
3              CHN         22                Beijing        Beijing   
4              USA         IL  Springfield, Illinois      Champaign   
...            ...        ...                    ...            ...   
66363          CHN         22                Beijing        Beijing   
66364          HRV         15                  Split          Split   
66365          NaN        NaN                    NaN            NaN   
66366          USA         CA            SF Bay Area  San Francisco   
66367          NaN        NaN                    NaN            NaN   

       founded_at  
0             NaN  
1      04-09-2014  
2             NaN  
3      01-01-2007  
4      01-01-2010  
...           ...  
66363         NaN  
66364  13-05-2012  
66365  01-01-2011  
66366  01-01-2014  
66367         NaN  

[66368 rows x 10 columns]

In [10]:
#print basic imformation like shape,info
company.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 66368 entries, 0 to 66367
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   permalink      66368 non-null  object
 1   name           66367 non-null  object
 2   homepage_url   61310 non-null  object
 3   category_list  63220 non-null  object
 4   status         66368 non-null  object
 5   country_code   59410 non-null  object
 6   state_code     57821 non-null  object
 7   region         58338 non-null  object
 8   city           58340 non-null  object
 9   founded_at     51147 non-null  object
dtypes: object(10)
memory usage: 5.1+ MB


In [11]:
# print basic imformation like shape,info
company.describe

<bound method NDFrame.describe of                                                permalink  \
0                                    /Organization/-Fame   
1                                 /Organization/-Qounter   
2                    /Organization/-The-One-Of-Them-Inc-   
3                                  /Organization/0-6-Com   
4                         /Organization/004-Technologies   
...                                                  ...   
66363  /Organization/Zznode-Science-And-Technology-Co...   
66364                          /Organization/Zzzzapp-Com   
66365                               /Organization/ÁEron   
66366                             /Organization/ÔAsys-2   
66367  /Organization/Ä°Novatiff-Reklam-Ve-Tanä±Tä±M-H...   

                                                name  \
0                                              #fame   
1                                           :Qounter   
2                             (THE) ONE of THEM,Inc.   
3                    

In [12]:
# print basic imformation like shape,info
company.shape

(66368, 10)

# print basic imformation like shape,info(Rounds)

In [13]:
rounds

permalink  \
0                                     /organization/-fame   
1                                  /ORGANIZATION/-QOUNTER   
2                                  /organization/-qounter   
3                     /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                                   /organization/0-6-com   
...                                                   ...   
114944                          /organization/zzzzapp-com   
114945                          /ORGANIZATION/ZZZZAPP-COM   
114946                               /organization/ãeron   
114947                             /ORGANIZATION/ÔASYS-2   
114948  /organization/ä°novatiff-reklam-ve-tanä±tä±m-h...   

                                funding_round_permalink funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1       /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2       /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3       /funding-round/650b8f704416801069bb178a1418776b            venture   
4       /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
...                                                 ...                ...   
114944  /funding-round/8f6d25b8ee4199e586484d817bceda05   convertible_note   
114945  /funding-round/ff1aa06ed5da186c84f101549035d4ae               seed   
114946  /funding-round/59f4dce44723b794f21ded3daed6e4fe            venture   
114947  /funding-round/35f09d0794651719b02bbfd859ba9ff5               seed   
114948  /funding-round/af942869878d2cd788ef5189b435ebc4              grant   

       funding_round_code   funded_at  raised_amount_usd  
0                       B  05-01-2015         10000000.0  
1                       A  14-10-2014                NaN  
2                     NaN  01-03-2014           700000.0  
3                       B  30-01-2014          3406878.0  
4                       A  19-03-2008          2000000.0  
...                   ...         ...                ...  
114944                NaN  01-03-2014            41313.0  
114945                NaN  01-05-2013            32842.0  
114946                  A  01-08-2014                NaN  
114947                NaN  01-01-2015            18192.0  
114948                NaN  01-10-2013            14851.0  

[114949 rows x 6 columns]

In [14]:
rounds.describe

<bound method NDFrame.describe of                                                 permalink  \
0                                     /organization/-fame   
1                                  /ORGANIZATION/-QOUNTER   
2                                  /organization/-qounter   
3                     /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                                   /organization/0-6-com   
...                                                   ...   
114944                          /organization/zzzzapp-com   
114945                          /ORGANIZATION/ZZZZAPP-COM   
114946                               /organization/ãeron   
114947                             /ORGANIZATION/ÔASYS-2   
114948  /organization/ä°novatiff-reklam-ve-tanä±tä±m-h...   

                                funding_round_permalink funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1       /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   

In [15]:
rounds.info

<bound method DataFrame.info of                                                 permalink  \
0                                     /organization/-fame   
1                                  /ORGANIZATION/-QOUNTER   
2                                  /organization/-qounter   
3                     /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                                   /organization/0-6-com   
...                                                   ...   
114944                          /organization/zzzzapp-com   
114945                          /ORGANIZATION/ZZZZAPP-COM   
114946                               /organization/ãeron   
114947                             /ORGANIZATION/ÔASYS-2   
114948  /organization/ä°novatiff-reklam-ve-tanä±tä±m-h...   

                                funding_round_permalink funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1       /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2

In [16]:
rounds.shape

(114949, 6)

In [17]:
#convert all the permalink column values into lowercase in both dataframes
x=company.permalink.str.lower()
x

0                                      /organization/-fame
1                                   /organization/-qounter
2                      /organization/-the-one-of-them-inc-
3                                    /organization/0-6-com
4                           /organization/004-technologies
                               ...                        
66363    /organization/zznode-science-and-technology-co...
66364                            /organization/zzzzapp-com
66365                                 /organization/ãeron
66366                               /organization/ã”asys-2
66367    /organization/ä°novatiff-reklam-ve-tanä±tä±m-h...
Name: permalink, Length: 66368, dtype: object

In [18]:
#convert all the permalink column values into lowercase in both dataframes
y=rounds.permalink.str.lower()
y

0                                       /organization/-fame
1                                    /organization/-qounter
2                                    /organization/-qounter
3                       /organization/-the-one-of-them-inc-
4                                     /organization/0-6-com
                                ...                        
114944                            /organization/zzzzapp-com
114945                            /organization/zzzzapp-com
114946                                 /organization/ãeron
114947                               /organization/ã”asys-2
114948    /organization/ä°novatiff-reklam-ve-tanä±tä±m-h...
Name: permalink, Length: 114949, dtype: object

In [19]:
#check how many unique values in permalink columns in both the dataframes
unique_values_company= company.nunique()
unique_values_company

permalink        66368
name             66102
homepage_url     61191
category_list    27296
status               4
country_code       137
state_code         311
region            1092
city              5111
founded_at        3978
dtype: int64

In [20]:
#check how many unique values in permalink columns in both the dataframes
unique_values_rounds= company.nunique()
unique_values_rounds

permalink        66368
name             66102
homepage_url     61191
category_list    27296
status               4
country_code       137
state_code         311
region            1092
city              5111
founded_at        3978
dtype: int64

In [21]:
# merge two dataframes into one master frame (on permalink)
add_company_rounds=[company,rounds]   
add_company_rounds

[                                               permalink  \
 0                                    /Organization/-Fame   
 1                                 /Organization/-Qounter   
 2                    /Organization/-The-One-Of-Them-Inc-   
 3                                  /Organization/0-6-Com   
 4                         /Organization/004-Technologies   
 ...                                                  ...   
 66363  /Organization/Zznode-Science-And-Technology-Co...   
 66364                          /Organization/Zzzzapp-Com   
 66365                               /Organization/ÁEron   
 66366                             /Organization/ÔAsys-2   
 66367  /Organization/Ä°Novatiff-Reklam-Ve-Tanä±Tä±M-H...   
 
                                                 name  \
 0                                              #fame   
 1                                           :Qounter   
 2                             (THE) ONE of THEM,Inc.   
 3                                    

In [22]:
from functools import reduce
data_merge=reduce(lambda left, right:
                 pd.merge(left,right,how="outer" ),add_company_rounds)
print(data_merge)

                                                permalink  \
0                                     /Organization/-Fame   
1                                  /Organization/-Qounter   
2                     /Organization/-The-One-Of-Them-Inc-   
3                                   /Organization/0-6-Com   
4                          /Organization/004-Technologies   
...                                                   ...   
181312                          /ORGANIZATION/ZZZZAPP-COM   
181313                          /ORGANIZATION/ZZZZAPP-COM   
181314                               /organization/ãeron   
181315                             /ORGANIZATION/ÔASYS-2   
181316  /organization/ä°novatiff-reklam-ve-tanä±tä±m-h...   

                          name                       homepage_url  \
0                        #fame                 http://livfame.com   
1                     :Qounter             http://www.qounter.com   
2       (THE) ONE of THEM,Inc.                http://oneofth

# masterframe will have around 114949 rows, If not please recheck the code.         

ROWS: 181317 ; COLUMNS: 15

In [23]:
# check percentage of missing values in each column
company["permalink"].isnull().sum()/len(company["permalink"])

0.0

In [25]:
# check percentage of missing values in each column
company["name"].isnull().sum()/len(company["name"])

1.5067502410800386e-05

In [26]:
# check percentage of missing values in each column
company["homepage_url"].isnull().sum()/len(company["homepage_url"])

0.07621142719382835

In [27]:
# check percentage of missing values in each column
company["category_list"].isnull().sum()/len(company["category_list"])

0.04743249758919962

In [28]:
# check percentage of missing values in each column
company["status"].isnull().sum()/len(company["status"])

0.0

In [29]:
# check percentage of missing values in each column
company["country_code"].isnull().sum()/len(company["country_code"])

0.10483968177434909

In [30]:
# check percentage of missing values in each column
company["state_code"].isnull().sum()/len(company["state_code"])

0.1287819431051109

In [31]:
# check percentage of missing values in each column
company["region"].isnull().sum()/len(company["region"])

0.1209920443587271

In [32]:
# check percentage of missing values in each column
company["city"].isnull().sum()/len(company["city"])

0.1209619093539055

In [33]:
# check percentage of missing values in each column
company["founded_at"].isnull().sum()/len(company["founded_at"])

0.22934245419479268

In [34]:
# check percentage of missing values in each column
rounds["permalink"].isnull().sum()/len(rounds["permalink"])

0.0

In [35]:
# check percentage of missing values in each column
rounds["funding_round_permalink"].isnull().sum()/len(rounds["funding_round_permalink"])

0.0

In [36]:
# check percentage of missing values in each column
rounds["funding_round_type"].isnull().sum()/len(rounds["funding_round_type"])

0.0

In [37]:
# check percentage of missing values in each column
rounds["funding_round_code"].isnull().sum()/len(rounds["funding_round_code"])

0.7290972518247223

In [38]:
# check percentage of missing values in each column
rounds["funded_at"].isnull().sum()/len(rounds["funded_at"])

0.0

In [39]:
# check percentage of missing values in each column
rounds["raised_amount_usd"].isnull().sum()/len(rounds["raised_amount_usd"])

0.17390320924931926

In [40]:
# drop all the rows where 'raised_amount_usd' value is zero.
rounds = rounds[rounds.raised_amount_usd.isin(rounds) == False]
rounds

permalink  \
0                                     /organization/-fame   
1                                  /ORGANIZATION/-QOUNTER   
2                                  /organization/-qounter   
3                     /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                                   /organization/0-6-com   
...                                                   ...   
114944                          /organization/zzzzapp-com   
114945                          /ORGANIZATION/ZZZZAPP-COM   
114946                               /organization/ãeron   
114947                             /ORGANIZATION/ÔASYS-2   
114948  /organization/ä°novatiff-reklam-ve-tanä±tä±m-h...   

                                funding_round_permalink funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1       /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2       /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3       /funding-round/650b8f704416801069bb178a1418776b            venture   
4       /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
...                                                 ...                ...   
114944  /funding-round/8f6d25b8ee4199e586484d817bceda05   convertible_note   
114945  /funding-round/ff1aa06ed5da186c84f101549035d4ae               seed   
114946  /funding-round/59f4dce44723b794f21ded3daed6e4fe            venture   
114947  /funding-round/35f09d0794651719b02bbfd859ba9ff5               seed   
114948  /funding-round/af942869878d2cd788ef5189b435ebc4              grant   

       funding_round_code   funded_at  raised_amount_usd  
0                       B  05-01-2015         10000000.0  
1                       A  14-10-2014                NaN  
2                     NaN  01-03-2014           700000.0  
3                       B  30-01-2014          3406878.0  
4                       A  19-03-2008          2000000.0  
...                   ...         ...                ...  
114944                NaN  01-03-2014            41313.0  
114945                NaN  01-05-2013            32842.0  
114946                  A  01-08-2014                NaN  
114947                NaN  01-01-2015            18192.0  
114948                NaN  01-10-2013            14851.0  

[114949 rows x 6 columns]

In [41]:
# drop funding_round_code,funded_at columns

rounds

permalink  \
0                                     /organization/-fame   
1                                  /ORGANIZATION/-QOUNTER   
2                                  /organization/-qounter   
3                     /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                                   /organization/0-6-com   
...                                                   ...   
114944                          /organization/zzzzapp-com   
114945                          /ORGANIZATION/ZZZZAPP-COM   
114946                               /organization/ãeron   
114947                             /ORGANIZATION/ÔASYS-2   
114948  /organization/ä°novatiff-reklam-ve-tanä±tä±m-h...   

                                funding_round_permalink funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1       /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2       /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3       /funding-round/650b8f704416801069bb178a1418776b            venture   
4       /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
...                                                 ...                ...   
114944  /funding-round/8f6d25b8ee4199e586484d817bceda05   convertible_note   
114945  /funding-round/ff1aa06ed5da186c84f101549035d4ae               seed   
114946  /funding-round/59f4dce44723b794f21ded3daed6e4fe            venture   
114947  /funding-round/35f09d0794651719b02bbfd859ba9ff5               seed   
114948  /funding-round/af942869878d2cd788ef5189b435ebc4              grant   

       funding_round_code   funded_at  raised_amount_usd  
0                       B  05-01-2015         10000000.0  
1                       A  14-10-2014                NaN  
2                     NaN  01-03-2014           700000.0  
3                       B  30-01-2014          3406878.0  
4                       A  19-03-2008          2000000.0  
...                   ...         ...                ...  
114944                NaN  01-03-2014            41313.0  
114945                NaN  01-05-2013            32842.0  
114946                  A  01-08-2014                NaN  
114947                NaN  01-01-2015            18192.0  
114948                NaN  01-10-2013            14851.0  

[114949 rows x 6 columns]

In [42]:
# aggregate rows with funding_type and print mean investment for each type
# you can use grouping
rounds.set_index(['funding_round_type']).mean

<bound method NDFrame._add_numeric_operations.<locals>.mean of                                                             permalink  \
funding_round_type                                                      
venture                                           /organization/-fame   
venture                                        /ORGANIZATION/-QOUNTER   
seed                                           /organization/-qounter   
venture                           /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
venture                                         /organization/0-6-com   
...                                                               ...   
convertible_note                            /organization/zzzzapp-com   
seed                                        /ORGANIZATION/ZZZZAPP-COM   
venture                                          /organization/ãeron   
seed                                           /ORGANIZATION/ÔASYS-2   
grant               /organization/ä°novatiff-reklam-ve-tanä±t

In [72]:
# find out which funding_type have more funding. the retain the rows of
# that paticular type and delete other rows. 
rounds["raised_amount_usd"].value_counts()

#rounds["funding_round_type"].value_counts()

1000000.0     2860
2000000.0     2225
500000.0      2145
5000000.0     2057
10000000.0    2052
              ... 
9259078.0        1
2665127.0        1
353102.0         1
782660.0         1
14851.0          1
Name: raised_amount_usd, Length: 22095, dtype: int64

In [115]:
#rounds.drop(["raised_amount_usd"!=1000000.0],axis=1)
drop_raised_amount_usd=rounds[rounds.raised_amount_usd =="10000000.0"]
#print(drop_raised_amount_usd)
print([rounds.raised_amount_usd==10000000.0])

[0          True
1         False
2         False
3         False
4         False
          ...  
114944    False
114945    False
114946    False
114947    False
114948    False
Name: raised_amount_usd, Length: 114949, dtype: bool]


In [120]:
result_df=  rounds.loc[rounds['raised_amount_usd']==10000000.0]
result_df

permalink  \
0                      /organization/-fame   
52                   /organization/1006-tv   
133                 /ORGANIZATION/17-MEDIA   
141                  /ORGANIZATION/17ZUOYE   
190     /organization/1st-merchant-funding   
...                                    ...   
114711                /ORGANIZATION/ZPOWER   
114742                 /organization/zubie   
114799       /ORGANIZATION/ZUNIVERSITY-COM   
114861       /ORGANIZATION/ZYGA-TECHNOLOGY   
114908                 /organization/zynga   

                                funding_round_permalink funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
52      /funding-round/b6aeb7401ec6993f92a16cbca153b600            venture   
133     /funding-round/f8ffde9fa822843ac6ea9b66cc615df1            venture   
141     /funding-round/8d87f771e938e0f31641bd600abbafca            venture   
190     /funding-round/ed21cc0e56104c9e20611730a5ebefc1     debt_financing   
...                                                 ...                ...   
114711  /funding-round/a8d010b9de7c04c98dad65033d381a40            venture   
114742  /funding-round/c834914cc13644c86b33542a0622ee8e            venture   
114799  /funding-round/a058abef15e17654e754acd8d4303af8            venture   
114861  /funding-round/00d2d8d0c604892594b4bd25cf7e18b8            venture   
114908  /funding-round/66d64809546fe2031d3479161e3c88f9            venture   

       funding_round_code   funded_at  raised_amount_usd  
0                       B  05-01-2015         10000000.0  
52                      B  31-07-2014         10000000.0  
133                     A  05-11-2015         10000000.0  
141                     B  01-09-2013         10000000.0  
190                   NaN  07-05-2014         10000000.0  
...                   ...         ...                ...  
114711                NaN  28-05-2014         10000000.0  
114742                  A  12-12-2013         10000000.0  
114799                  B  10-01-2001         10000000.0  
114861                  C  16-12-2013         10000000.0  
114908                  A  01-01-2008         10000000.0  

[2052 rows x 6 columns]

In [143]:
URL = "https://en.wikipedia.org/wiki/List_of_territorial_entities_where_English_is_an_official_language"
# use above link and extract all english( as primary language) speaking countries. 
'''
#-------------------this code will help you to get countries table
tables = pd.read_html(URL)
table=tables[1]
countries=pd.DataFrame(table)

you manually need to clean table for better understanding
'''
english_language_speaking=pd.read_html("https://en.wikipedia.org/wiki/List_of_territorial_entities_where_English_is_an_official_language")
english_language_speaking

[                                 Country ISO code      Geographic region  \
 0                    Antigua and Barbuda      ATG              Caribbean   
 1                         The Bahamas[1]      BHS              Caribbean   
 2                            Barbados[2]      BRB              Caribbean   
 3                              Belize[3]      BLZ        Central America   
 4                            Botswana[3]      BWA                 Africa   
 5                             Burundi[4]      BDI                 Africa   
 6                            Cameroon[1]      CMR                 Africa   
 7                                 Canada      CAN          North America   
 8                            Dominica[1]      DMA              Caribbean   
 9                            Eswatini[1]      SWZ                 Africa   
 10                               Fiji[1]      FJI                Oceania   
 11                         The Gambia[1]      GMB                 Africa   

In [241]:
#print(drop_raised_amount_usd)
#print([english_language_speaking.Country.head(3)])
import matplotlib.pyplot as plt
import pandas as pd
english_language_speaking.Country.value_counts()

Australia          1
New Zealand[24]    1
United Kingdom     1
United States      1
Name: Country, dtype: int64

In [144]:
'''
#-------------------this code will help you to get countries table
tables = pd.read_html(URL)
table=tables[1]
countries=pd.DataFrame(table)

you manually need to clean table for better understanding
'''
english_language_speaking=english_language_speaking[1]
english_language_speaking_countries=pd.DataFrame(english_language_speaking)
english_language_speaking_countries

Country ISO code Geographic region  Population1 Primary language?
0        Australia      AUS           Oceania     25795700               Yes
1  New Zealand[24]      NZL           Oceania      4893830               Yes
2   United Kingdom      GBR            Europe     66040229               Yes
3    United States      USA     North America    328239523               Yes

In [172]:
# now group data by country code (as per country code is given dataframe) 
# find out top 10 countries with highest funding. (raised_amount_usd is the column to aggregate)
"""result_df =  english_language_speaking.loc[english_language_speaking['Country']==english]
result_df  """
from pandas.api.types import is_numeric_dtype

english_language_speaking_countries = []

for raised_amount_usd in english_language_speaking.columns:
    top_values = []
    if is_numeric_dtype(english_language_speaking[raised_amount_usd]):
        top_values = english_language_speaking[raised_amount_usd].nlargest(n=10)
        english_language_speaking_countries.append(pd.DataFrame({raised_amount_usd: top_values}).reset_index(drop=True))
pd.concat(english_language_speaking_countries, axis=1)


Population1
0    328239523
1     66040229
2     25795700
3      4893830

In [209]:
# find out top 3 english speaking countries to invest.(based on raised amount of investment)
# you should get. [United States of America , India, United Kingdom]
english_language_speaking.value_counts()

Country          ISO code  Geographic region  Population1  Primary language?
Australia        AUS       Oceania            25795700     Yes                  1
New Zealand[24]  NZL       Oceania            4893830      Yes                  1
United Kingdom   GBR       Europe             66040229     Yes                  1
United States    USA       North America      328239523    Yes                  1
dtype: int64

In [1]:
abc=english_language_speaking[english_language_speaking.Country =="Population1"]
#print(drop_raised_amount_usd)
#print([english_language_speaking.Country.head(3)])
import matplotlib.pyplot as plt
import pandas as pd
temp = english_language_speaking.groupby(['Country']).sum()['Population1']

plt.pie(temp.values,labels=temp.index,colors=['red','green','blue','orange','black','brown'])

plt.show()

NameError: name 'english_language_speaking' is not defined

# Plese explain how would you split investment percentage amoung these countries and why
# this question is understand your business prespective.
"""sir, Im srikanth sri G-17 python. here we are having manyways to find how to split 
investment percentage amoung these countries. I thought to do by visualizing the data.
so i used the above link and extracted all english speaking countries. Now i tried to clean 
some data with that i will able to understand the data properly.after that we can observe heighest funding
countries.As you asked top 3 english speaking countries to invest i thought to do By showing in pie chart.
"""

# how do you decide, What value should be imputed in missing place.
# this is subjective question. write your own answer based on what you understood.
"""Its upon the data what we have recieved. some times we can fill some null values for understanding purpose.
but it willnot work every time. as we observe some bank data sets, data will be different for differnet persons.
banker wants to know what kind of people he should  approach so that he can make more people sign up for personal
loans using the data he collected. so data is very important but the way someone visualise the data is the 
important thing. so according to data we have to imputed in missing places"""